pkg/appmanager/profiles.go

```diff
@@ -426,9 +440,15 @@ func (appMgr *Manager) deleteUnusedProfiles(
 
 		var toRemove []ProfileRef
 		for _, prof := range cfg.Virtual.Profiles {
+
+			// If reencryptRouteCount >= 1, don't remove default-route-serverssl
+			if reencryptRouteCount >= 1 && prof.Name == "default-route-serverssl" {
+				reencryptRouteCount = 0
+				continue
+			}
 			// Don't process profiles that came from a resource in a different namespace.
 			// We don't want to delete them, since they won't reference a resource this time.
-			if prof.Namespace != namespace {
+			if prof.Namespace != namespace && prof.Name != "default-route-serverssl" {
```

Comment:
Since we don't care about resources in other NS, why should you check the name ?

Reply:
This is needed. This method iterates over all the profiles, since the default profiles does not have any namespaces, these profiles are ignored for deletion.